### PR TITLE
refactor: Replace Badge with rc-motion

### PR DESCRIPTION
**Pull request by zombieJ**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [x] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |   Replace Badge with `rc-motion` instead of `rc-animate`.        |
| 🇨🇳 Chinese |     替换 Badge 所使用的 `rc-animate` 动画库为 `rc-motion`。     |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

### CODE REVIEW

File: components/badge/ScrollNumber.tsx [hunk 1]
```diff
@@ -142,21 +154,15 @@ const ScrollNumber: React.FC<ScrollNumberProps> = ({
...
-    return count;
-  };
-
-  const newProps = {
-    ...restProps,
-    style,
-    className: classNames(prefixCls, className),
-    title: title as string,
-  };
+  const numberNodeRef = React.useRef<React.ReactNode>(null);
```
Comment (hengkx) [inline components/badge/ScrollNumber.tsx:157]: 这个 直接 写明 具体 类型 会不会 更好点

Comment (zombieJ) [reply components/badge/ScrollNumber.tsx:157]: `string | number | null | undefined | React.ReactElement[]` 基本和 `ReactNode` 等价了。

Comment (hengkx) [reply components/badge/ScrollNumber.tsx:157]: 好吧 没进方法看，以为只有 string 和 number

**Review by hengkx — APPROVED** [review]

File: components/badge/__tests__/__snapshots__/index.test.js.snap [hunk 2]
```diff
@@ -1413,39 +1413,25 @@ exports[`Badge should render when count is changed 1`] = `
...
-      showProp="data-show"
-      transitionAppear={true}
-      transitionEnter={true}
-      transitionLeave={true}
-      transitionName=""
+    <CSSMotion
+      motionAppear={true}
+      motionName="ant-badge-zoom"
+      visible={true}
     >
```
Comment (afc163) [inline components/badge/__tests__/__snapshots__/index.test.js.snap:1420]: 这个也顺便改成 render snapshot 好了。

**Review by afc163 — APPROVED** [review]

**Pull request merged**